tests/test_homework/test_emailSender.py

```diff
@@ -0,0 +1,173 @@
+"""
+
+import pytest
+from homework.emailSender import EmailSender, ReportFileError
+
+
+@pytest.fixture()
+def set_monkey_attrs_n_envs(monkeypatch, tmp_path):
+    """Create fixture instance of EmailSender."""
+
+    class MockLoginSession:
+        """Simulate _start_login_session.
+
+        This is a replacement for self.session = self._start_login_session().
+        Instead of creating external SMTP object, we only simulate it.
+        """
+
+        def __init__(self):
+            pass
+
+        def sendmail(self, *_, **__):  # args and kwargs
+            """Simulate self.session.sendmail"""
+            pass
+
+        def quit(self, *_, **__):  # args and kwargs
+            """Simulate self.session.quit()"""
+            pass
+
+    template_path = tmp_path / "path.txt"
+    template_path.write_text("Text message from template path")
+    monkeypatch.setenv("SENDER_EMAIL", "sender@email.com")
+    monkeypatch.setenv("SENDER_PASSWORD", "secretPassword.!")
+    monkeypatch.setenv("EMAIL_TEMPLATE_PATH", str(template_path))
+    monkeypatch.setattr(
+        "homework.emailSender.EmailSender._start_login_session", MockLoginSession
+    )
+
+
+@pytest.fixture()
+def email_sender(set_monkey_attrs_n_envs):
+    """Create EmailSender instance in fixture."""
+
+    sender = EmailSender(
+        subject="Subject Text",
+        to_email="receiver@mail.com",
+        cc_emails=["cc_receiver_1@mail.com", "cc_receiver_2@mail.com"],
+    )
+    return sender
+
+
+@pytest.mark.parametrize(
+    "input_message, output",
+    [
+        ("Given text message", "Given text message"),
+        (None, "Text message from template path"),
+    ],
+)
+def test_create_email_sender_instance(input_message, output, set_monkey_attrs_n_envs):
+    """Check if EmailSender object was created with every attribute."""
+
+    parametrized_sender = EmailSender(
+        subject="Subject Text",
+        to_email="receiver@mail.com",
+        cc_emails=["cc_receiver_1@mail.com", "cc_receiver_2@mail.com"],
+        email_message=input_message,
+    )
+
+    assert isinstance(parametrized_sender, EmailSender)
+    assert output in parametrized_sender.email_message
+    assert (
+        "cc_receiver_1@mail.com,cc_receiver_2@mail.com" in parametrized_sender.cc_emails
+    )
+    assert "sender@email.com" in parametrized_sender.sender_email
+
+
+def test_check_file_text_format(set_monkey_attrs_n_envs, tmp_path, monkeypatch):
+    """Check if EMAIL_TEMPLATE_PATH env has a proper format - .txt"""
+
+    template_path = tmp_path / "path.docx"
+    monkeypatch.setenv("EMAIL_TEMPLATE_PATH", str(template_path))
+    with pytest.raises(
+        ReportFileError, match="Email path env should lead to .txt file!"
+    ):
+        EmailSender(
+            subject="Subject Text",
+            to_email="receiver@mail.com",
+            cc_emails=["cc_receiver_1@mail.com", "cc_receiver_2@mail.com"],
+        )
+
+
+def test_create_email(email_sender, monkeypatch, tmp_path):
```

Comment:
To jest....wow! Tylko może doałabym tu komentarz nad tym, że to rozwiąznie dla "chętnych" bo faktycznie hu hu hu :D 💙